src/services/identity/SitesCacheService.ts

```diff
@@ -14,81 +16,225 @@ import type { GitHubRepositoryData, RepositoryData } from "@root/types/repoInfo"
 // Changes are polled at fixed intervals, stored in cache, and served
 // to avoid long load time from live querying.
 
+const DEFAULT_PAGE_PARAMS = {
+  per_page: GH_MAX_REPO_COUNT,
+  sort: "full_name",
+  page: 1,
+}
+
+type GitHubResponseHeaders = HeadersDefaults & {
+  link?: string
+}
+
+type LinkRelation = "first" | "last" | "prev" | "next"
+
+type Link = {
+  relid: LinkRelation
+  url: string
+  pagenum: number
+}
+
+type LinkSet = {
+  [relid in LinkRelation]?: Link
+}
+
+type LinkMatch = RegExpExecArray & {
+  groups: {
+    relid: LinkRelation
+    url: string
+  }
+}
+
+type FetchRepoPageResult = {
+  repos: RepositoryData[]
+  links?: LinkSet
+}
+
+type CacheStore = {
+  [key: string]: RepositoryData
+}
+
+function parseGitHubLinkHeader(linkheader: string): LinkSet {
+  // example value: link: <https://api.github.com/organizations/40887764/repos?page=2>; rel="next", <https://api.github.com/organizations/40887764/repos?page=34>; rel="last"
+  const links: LinkSet = {}
+
+  const linkRe = /<(?<url>[^>]+)>; rel="(?<relid>[a-z]+)"(, )?/g
```

Comment:
I'd rather leave the regexp loose tbh, in case github introduces more link relations in the future, they'd already be captured properly as part of the parsing routine. To be consistent then, we could upgrade the type to allow unknown keys too 🤔